### DISCUSSION TITLE
[@types/select2] ajax missing jQuery.ajax types

### DISCUSSION THREAD

**Discussion by kkmuffme** [DefinitelyTyped/DefinitelyTyped#58036] [Issues with a @types package]
select2 ajax extends the jQuery.ajax method (https://select2.org/data-sources/ajax#jquery-ajax-options)
but the stubs have a bug in them, that it does not correctly extend that.

```
jQuery('.js-data-example-ajax').select2({
	ajax: {
		url: 'https://api.github.com/search/repositories',
		delay: 250 
		dataType: 'json'
		// Additional AJAX parameters go here; see the end of this chapter for the full code of this example
	}
});
```

e.g. dataType is part of jQuery.ajax https://api.jquery.com/jquery.ajax/
but it's missing in the stubs, thus getting it reported as error incorrectly

[a comment by an automated account is not shown]

**Comment by martin-badin** [comment]
Hi. I added a few tests and it seems to look good. #58052
